fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordKeyCursorTest.java

```diff
@@ -0,0 +1,270 @@
+            for (int i = 0; i < 20; i++) {
+                Tuple pkAndVersion = savedRecords.get(i).getPrimaryKey().add(-1);
+                byte[] versionKey = store.recordsSubspace().pack(pkAndVersion);
+                store.ensureContextActive().clear(versionKey);
+            }
+            commit(context);
+        }
+
+        // Scan records
+        ScanProperties scanProperties = getScanProperties(useContinuations);
+        final List<Tuple> actualKeys;
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION);
+            actualKeys = scanKeys(store, scanProperties, useContinuations != UseContinuations.NONE);
+            context.commit();
+        }
+        List<Tuple> expectedKeys = IntStream.range(1, 101).boxed().map(Tuple::from).collect(Collectors.toList());
+        assertEquals(expectedKeys, actualKeys);
+    }
+
+    @Nonnull
+    private static RecordMetaDataHook getRecordMetaDataHook(final boolean splitLongRecords) {
+        return metaData -> {
+            metaData.setSplitLongRecords(splitLongRecords);
+            // index cannot be used with large fields
+            metaData.removeIndex("MySimpleRecord$str_value_indexed");
+        };
+    }
+
+    @Nullable
+    private static ScanProperties getScanProperties(final UseContinuations useContinuations) {
+        ExecuteProperties executeProperties;
+        switch (useContinuations) {
+            case NONE:
+                return null;
+            case CONTINUATIONS:
+                executeProperties = ExecuteProperties.newBuilder()
+                        .setReturnedRowLimit(19)
+                        .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                        .build();
+                return new ScanProperties(executeProperties);
+            case BYTE_LIMIT:
+                executeProperties = ExecuteProperties.newBuilder()
+                        .setScannedBytesLimit(2000)
+                        .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                        .build();
+                return new ScanProperties(executeProperties);
+            default:
+                throw new IllegalArgumentException("Unknown continuation");
+        }
+    }
+
+    private List<Tuple> scanKeys(@Nonnull FDBRecordStore store, @Nullable ScanProperties scanProperties, boolean withContinuations) throws InterruptedException, ExecutionException {
+        if (scanProperties == null) {
+            scanProperties = ScanProperties.FORWARD_SCAN;
+        }
+        RecordCursor<Tuple> recordKeyCursor = store.scanRecordKeys(TupleRange.allOf(null), null, scanProperties);
+        if (!withContinuations) {
+            return recordKeyCursor.asList().get();
+        } else {
+            boolean done = false;
+            List<Tuple> result = new ArrayList<>();
+            while (!done) {
```

Comment:
This should probably assert that the limits are respected. Right now, if they're ignored, it will still succeed, and also lose all of the testing of continuations. The bytes limit doesn't need precise assertions here, but making sure that the result has the correct NoNextReason would be good, and that it wasn't all of them in the first try.

Reply:
Done.